sc2/client.py

```diff
@@ -118,7 +120,46 @@ async def leave(self):
         except (ProtocolError, ConnectionAlreadyClosed):
             if is_resign:
                 raise
+    
+    async def unload_unit(self, transporter_unit: Unit, cargo_unit: Unit = False):
+        """ 
+            Unloads single unit passed by cargo_unit or first unit in transporter if not cargo_unit passed 
+            transporter_unit includes all units, which can unload cargo: 
+            warp prism, medivac, bunker, command center, command center flying, planetary fortress, droppenlord, nydlus
+            
+            Usage:
+            self.client.unload_unit(transporter_unit, cargo_unit)
+            self.client.unload_unit(transporter_unit) # unloads first one
+
+        """
+        assert isinstance(transporter_unit, Unit)
+        assert isinstance(cargo_unit, (bool, Unit))
+           
+        if not transporter_unit.passengers:
+            return
+
+        if isinstance(cargo_unit, bool):
+            unload_unit_index = 0
+        if isinstance(cargo_unit, Unit):
+            unload_unit_index = next((index for index, unit in enumerate(transporter_unit._proto.passengers) if unit.tag == cargo_unit.tag), 0)
 
+        await self._execute(
+            action=sc_pb.RequestAction(
+                actions=[
+                    sc_pb.Action(
+                        action_raw=raw_pb.ActionRaw(
+                            unit_command=raw_pb.ActionRawUnitCommand(ability_id=0, unit_tags=[transporter_unit.tag])
+                        )
+                    ),
+                    sc_pb.Action(
+                        action_ui=ui_pb.ActionUI(
+                            cargo_panel=ui_pb.ActionCargoPanelUnload(unit_index=unload_unit_index)
+                        )
+                    ),
+                ]
+            )
+        )
```

Comment:
This executes the command instantly. Is there a better way to bundle it with other unit commands?
Currently actions such as `unit.move(position)` and `structure.train(unittype)` are collected and sent once to the API at the end of the step. Is it also possible with this?

Reply:
It would be rare situation where someone use more than one unload command in same frame over a game. 
The most usage I have seen is a bot that has 4 prism and with human eye has uploads aren't synchronized.